controllers/stargate_controller_test.go

```diff
@@ -96,50 +106,245 @@ func testCreate(t *testing.T, testClient client.Client) {
 	dc.Status.UsersUpserted = metav1.Now()
 
 	err = testClient.Status().Update(ctx, dc)
-	req.NoError(err, "failed to update dc")
+	require.NoError(t, err, "failed to update dc")
 
 	err = testClient.Create(ctx, stargate)
-	req.NoError(err, "failed to create Stargate")
+	require.NoError(t, err, "failed to create Stargate")
 
 	t.Log("check that the Stargate resource was created")
 	stargateKey := types.NamespacedName{Namespace: namespace, Name: "dc1-stargate"}
-	req.Eventually(func() bool {
+	require.Eventually(t, func() bool {
 		err := testClient.Get(ctx, stargateKey, stargate)
 		return err == nil && stargate.Status.Progress == api.StargateProgressDeploying
 	}, timeout, interval)
 
-	deploymentKey := types.NamespacedName{Namespace: namespace, Name: "test-dc1-stargate-deployment"}
+	deploymentKey := types.NamespacedName{Namespace: namespace, Name: "test-dc1-default-stargate-deployment"}
 	deployment := &appsv1.Deployment{}
-	req.Eventually(func() bool {
+	require.Eventually(t, func() bool {
 		err := testClient.Get(ctx, deploymentKey, deployment)
 		return err == nil
 	}, timeout, interval)
 
 	t.Log("check that the owner reference is set on the Stargate deployment")
-	req.Len(deployment.OwnerReferences, 1, "expected to find 1 owner reference for Stargate deployment")
-	req.Equal(stargate.UID, deployment.OwnerReferences[0].UID)
+	require.Len(t, deployment.OwnerReferences, 1, "expected to find 1 owner reference for Stargate deployment")
+	require.Equal(t, stargate.UID, deployment.OwnerReferences[0].UID)
 
 	deployment.Status.Replicas = 1
 	deployment.Status.ReadyReplicas = 1
 	deployment.Status.AvailableReplicas = 1
+	deployment.Status.UpdatedReplicas = 1
 	err = testClient.Status().Update(ctx, deployment)
-	req.NoError(err, "failed to update deployment")
+	require.NoError(t, err, "failed to update deployment")
 
 	serviceKey := types.NamespacedName{Namespace: namespace, Name: "test-dc1-stargate-service"}
 	service := &corev1.Service{}
-	req.Eventually(func() bool {
+	require.Eventually(t, func() bool {
+		err := testClient.Get(ctx, serviceKey, service)
+		return err == nil
+	}, timeout, interval)
+
+	t.Log("check that the Stargate resource is fully reconciled")
+	require.Eventually(t, func() bool {
+		err := testClient.Get(ctx, stargateKey, stargate)
+		return err == nil && stargate.Status.Progress == api.StargateProgressRunning
+	}, timeout, interval)
+
+	t.Log("check Stargate status")
+	require.EqualValues(t, 1, stargate.Status.Replicas, "expected to find 1 replica for Stargate")
```

Comment:
minor nit - consider the `assert` equivalent functions that would allow for multiple verifications before failing.